source/slang/slang-ir-legalize-global-values.cpp

```diff
@@ -0,0 +1,233 @@
+#include "slang-ir-legalize-global-values.h"
+
+#include "slang-ir-clone.h"
+#include "slang-ir-util.h"
+
+namespace Slang
+{
+
+void GlobalInstInliningContextGeneric::inlineGlobalValuesAndRemoveIfUnused(IRModule* module)
+{
+    Dictionary<IRInst*, List<IRUse*>> globalInstUsesToInline;
+
+    for (auto globalInst : module->getGlobalInsts())
+    {
+        if (isInlinableGlobalInst(globalInst))
+        {
+            for (auto use = globalInst->firstUse; use; use = use->nextUse)
+            {
+                if (getParentFunc(use->getUser()) != nullptr)
+                {
+                    globalInstUsesToInline.getOrAddValue(globalInst, List<IRUse*>()).add(use);
+                }
+            }
+        }
+    }
+
+    for (auto globalInstUses : globalInstUsesToInline)
```

Comment:
this will also trigger a deep copy to get `globalInstUses`.

Reply:
The code should now be back to what it was. I think this is now just a pointer copy.